### PR TITLE
CR-1088690 CR-1111805 Report mem type information in validate designs

### PR DESCRIPTION
**Pull request by vishnuchebrolu**
https://jira.xilinx.com/browse/CR-1088690
https://jira.xilinx.com/browse/CR-1111805

Validate designs updated to report memory type while reporting throughput.

1. bandwidth_test
**For DDR memory :** 
Throughput (Type: DDR) (Bank count: 4): 52265 MB/s
**For HBM memory :** 
Throughput (Type: HBM) (Bank count: 1): 15328 MB/s
**For DDR and HBM memory :** 
Throughput (Type: DDR) (Bank count: 2): 29265 MB/s
Throughput (Type: HBM) (Bank count: 1): 15328 MB/s 
2. hostmemory_test:
**For HOST memory :** 
Throughput (Type: HOST) (Bank count: 1): 15328 MB/s

@AShivangi Please update the xbutil flow search string to pick the throughput numbers.

### CODE REVIEW

File: tests/validate/bandwidth_test/src/host.cpp [hunk 1]
```diff
@@ -66,26 +68,36 @@ int main(int argc, char** argv) {
...
+        auto pt_mem_array = load_ptree_root.get_child("meminfo");
+        for (const auto& mem_entry : pt_mem_array) {
+            boost::property_tree::ptree pt_mem_entry = mem_entry.second;
+            auto sValue = pt_mem_entry.get<std::string>("type");
+            if (sValue == "HBM") {
+                chk_hbm_mem = true;
+            }
+        }
+        if (chk_hbm_mem) {
+            num_kernel_ddr = num_kernel - 1;
```
Comment (rozumx) [inline tests/validate/bandwidth_test/src/host.cpp:92]: Please do add a comment why we need to subtract a 1.

File: tests/validate/bandwidth_test/src/host.cpp [hunk 2]
```diff
@@ -151,102 +161,175 @@ int main(int argc, char** argv) {
...
+            std::vector<unsigned char, aligned_allocator<unsigned char> > output_host[num_kernel_ddr];
 
-        if (xcl::is_emulation()) {
-            reps = 2;
-            if (DATA_SIZE > 8 * 1024) break;
-        }
+            for (int i = 0; i < num_kernel_ddr; i++) {
+                output_host[i].resize(data_size);
+            }
+            for (uint32_t j = 0; j < data_size; j++) {
```
Comment (rozumx) [inline tests/validate/bandwidth_test/src/host.cpp:184]: Comment here would be nice here as well.  Something like:
`// Filling up memory with an incremental byte pattern`

File: tests/validate/bandwidth_test/src/host.cpp [hunk 3]
```diff
@@ -151,102 +161,175 @@ int main(int argc, char** argv) {
...
-        }
+            std::vector<cl::Buffer> input_buffer(num_kernel_ddr);
+            std::vector<cl::Buffer> output_buffer(num_kernel_ddr);
+
+            // These commands will allocate memory on the FPGA. The cl::Buffer objects
+            // can
+            // be used to reference the memory locations on the device.
+            // Creating Buffers
+            for (int i = 0; i < num_kernel_ddr; i++) {
+                OCL_CHECK(err,
```
Comment (rozumx) [inline tests/validate/bandwidth_test/src/host.cpp:201]: Hmm...format is inconsistent with the rest of the method.  Might wish to put it back on one line for this isn't kernel C code.

**Review by rozumx — APPROVED** [review]
Approved with minor comment and format changes.

File: tests/validate/bandwidth_test/src/host.cpp (hunk 3 above)
Comment (vishnuchebrolu) [reply tests/validate/bandwidth_test/src/host.cpp:203]: We have a format.sh script that is used for code formatting. The host code pushed in PR is after running the format script.
/wrk/xhdhdnobkup1/hatchuta/public/format.sh

**Pull request merged**